invokeai/frontend/web/src/features/nodes/components/ui/ClearNodesButton.tsx

```diff
@@ -0,0 +1,86 @@
+import {
+  AlertDialog,
+  AlertDialogBody,
+  AlertDialogContent,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogOverlay,
+  Button,
+  Text,
+  useDisclosure,
+} from '@chakra-ui/react';
+import { makeToast } from 'app/components/Toaster';
+import { RootState } from 'app/store/store';
+import { useAppDispatch, useAppSelector } from 'app/store/storeHooks';
+import IAIIconButton from 'common/components/IAIIconButton';
+import { clearNodes } from 'features/nodes/store/nodesSlice';
+import { addToast } from 'features/system/store/systemSlice';
+import { memo, useRef } from 'react';
+import { useTranslation } from 'react-i18next';
+import { FaTrash } from 'react-icons/fa';
+
+const ClearNodesButton = () => {
+  const { t } = useTranslation();
+  const dispatch = useAppDispatch();
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const cancelRef = useRef<HTMLButtonElement | null>(null);
+
+  const nodes = useAppSelector((state: RootState) => state.nodes.nodes);
+
+  const handleConfirmClear = () => {
+    dispatch(clearNodes());
+
+    dispatch(
+      addToast(
+        makeToast({
+          title: t('toast.nodesCleared'),
+          status: 'success',
+        })
+      )
+    );
+
+    onClose();
+  };
```

Comment:
Please wrap this function in a `useCallback`, for example:
```ts
  const handleConfirmClear = useCallback(() => {
    dispatch(clearNodes());

    dispatch(
      addToast(
        makeToast({
          title: t('toast.nodesCleared'),
          status: 'success',
        })
      )
    );

    onClose();
  }, [dispatch, onClose, t]);
```

Reasoning:
Every time this component re-renders, the code in the component is run again. The `handleConfirmClear` function will be recreated every time.

In simpler applications (like a website), this is no big deal, but we have a ton of stuff happening, and need to be as efficient as possible. Every millisecond counts. 

By wrapping the function in `useCallback`, we tell react to not recreate this function on every render - it's maintained with a stable reference in memory. Each render then is slightly less resource-intensive.

The dependency array `[dispatch, onClose, t]` tells react that whenever one of those things changes, it should recreate the function, so it doesn't get stale. It's important to ensure that array is correct. VSCode gives me red squigglies if it's not correct, hopefully it does for you too.

Reply:
Ok, how do reopen branch thag was deleted or will i need to start a new pr